src/dfx-core/src/config/model/dfinity.rs

```diff
@@ -234,13 +234,14 @@ pub struct ConfigCanistersCanister {
 
     /// # Shrink Canister WASM
     /// Whether run `ic-wasm shrink` after building the Canister.
-    /// Enabled by default for Rust/Motoko canisters.
-    /// Disabled by default for custom canisters.
+    /// If not specify `optimize`, this option will be true by default.
+    /// To turn off all optimization/shrink, please explicitly set this option to false.
```

Comment:
We still shouldn't alter the wasm by default for custom canisters (including shrink).  We've gotten plenty of feedback about this.  Did something change?

Reply:
My main purpose is to make the behavior consistent for all types of canisters.
When I introduced `shrink`, I make it disable by default for custom canisters because the wasm may be gzipped.
In this PR, I force the custom canister to be non-gzipped. So the `shrink` is always viable.

I'm aware that developer may want to install the same canister specified in `dfx.json` without any modification. So here is a tradeoff, consistency v.s. no modification on custom canister wasm.

Another option is to make `shrink` disable by default for all canister types.

I unified the `optimize/shrink` code from builders into `step_postbuild`. I want to avoid adding another `if ` statement there.

Actually, I want to deprecate `shrink` and use `optimize` only. Then it will be no `optimize/shrink` by default. And devs can opt in `optimize`.